### PR TITLE
doc: update helm chart documentations

### LINKED ISSUE #1076: Update helm chart documentations
Current helm chart Readme https://github.com/ceph/ceph-csi/blob/master/charts/ceph-csi-cephfs/README.md and https://github.com/ceph/ceph-csi/tree/master/charts/ceph-csi-rbd  will talk about only on how to install and uninstall cephfs and rbd csi driver. we need to document all the available options when deploying these helm charts, This will help users to understand available options which they need to pass when there are installing helm charts

### PR DESCRIPTION
**Pull request by yati1998**
This commit updates the helm chart documentations
with the configurations available while deploying
these helm charts.

Signed-off-by: Yati Padia <ypadia@redhat.com>

### CODE REVIEW

**Comment by nixpanic** [comment]
@Yuggupta27 , #1786 adds extra options too. These need to be documented when this PR is merged. Maybe review this now, and rebase your PR once this is merged?

File: charts/ceph-csi-cephfs/README.md [hunk 1]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+
+### Configuration
+
+The following table lists the configurable parameters of the ceph-csi-cephfs
+charts and their default values.
+
+| Parameter | Description | Default |
+| ---------------------- | ---------------- | ---------------- |
+| `rbac.create` | Specifies whether RBAC resources should be created | `true` |
+| `serviceAccounts.nodeplugin.create` | Specifies whether a ServiceAccount should be created | `true` |
```
Comment (Yuggupta27) [inline charts/ceph-csi-cephfs/README.md]: This seems generic, please specify which ServiceAccount

File: charts/ceph-csi-cephfs/README.md [hunk 2]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `serviceAccounts.nodeplugin.name` | The name of the ServiceAccount to use. If not set and create is true, a name is generated using the fullname | "" |
+| `serviceAccounts.provisioner.create` |  Specifies whether a ServiceAccount should be created | `true` |
+| `serviceAccounts.provisioner.name` | The name of the ServiceAccount to use. If not set and create is true, a name is generated using the fullname | "" |
```
Comment (Yuggupta27) [inline charts/ceph-csi-cephfs/README.md]: same everywhere here.

File: charts/ceph-csi-cephfs/README.md [hunk 3]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `csiConfig` | Configuration for the CSI to connect to the cluster | [] |
+| `logLevel` | Set logging level for csi containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity. | `5` |
+| `nodeplugin.name` | Specifies the name | `nodeplugin` |
```
Comment (Yuggupta27) [inline charts/ceph-csi-cephfs/README.md]: `Specifies the name` seems general. Please specify which name.

File: charts/ceph-csi-cephfs/README.md [hunk 4]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `nodeplugin.updateStrategy` | Specifies the update Strategy. If you are using ceph-fuse client set this value to OnDelete | `RollingUpdate` |
+| `nodeplugin.priorityClassName` |  set user created priorityclassName for csi plugin pods. default is system-node-critical which is highest priority | `system-node-critical` |
+| `nodeplugin.profiling.enabled` | Specifies whether profiling should be enabled | `false` |
+| `nodeplugin.registrar.image.repository` | Image | `k8s.gcr.io/sig-storage/csi-node-driver-registrar` |
+| `nodeplugin.registrar.image.tag` | Image tag | `v2.2.0` |
+| `nodeplugin.registrar.image.pullPolicy` | Image pull policy | `IfNotPresent` |
+| `nodeplugin.plugin.image.repository` | Image | `quay.io/cephcsi/cephcsi` |
```
Comment (Yuggupta27) [inline charts/ceph-csi-cephfs/README.md]: `Image` doesn't seem to describe anything. Maybe `Image Repository`?

File: charts/ceph-csi-cephfs/README.md [hunk 5]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `serviceAccounts.provisioner.create` |  Specifies whether a ServiceAccount should be created | `true` |
+| `serviceAccounts.provisioner.name` | The name of the ServiceAccount to use. If not set and create is true, a name is generated using the fullname | "" |
+| `csiConfig` | Configuration for the CSI to connect to the cluster | [] |
+| `logLevel` | Set logging level for csi containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity. | `5` |
+| `nodeplugin.name` | Specifies the name | `nodeplugin` |
+| `nodeplugin.updateStrategy` | Specifies the update Strategy. If you are using ceph-fuse client set this value to OnDelete | `RollingUpdate` |
+| `nodeplugin.priorityClassName` |  set user created priorityclassName for csi plugin pods. default is system-node-critical which is highest priority | `system-node-critical` |
+| `nodeplugin.profiling.enabled` | Specifies whether profiling should be enabled | `false` |
+| `nodeplugin.registrar.image.repository` | Image | `k8s.gcr.io/sig-storage/csi-node-driver-registrar` |
```
Comment (Yuggupta27) [inline charts/ceph-csi-cephfs/README.md]: `Image` doesn't seem to describe anything. Maybe `Image Repository`?

File: charts/ceph-csi-cephfs/README.md [hunk 6]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `nodeplugin.plugin.image.tag` | Image tag | `canary` |
+| `nodeplugin.plugin.image.pullPolicy` | Image pull policy | `IfNotPresent` |
+| `nodeplugin.nodeSelector` | Kubernetes `nodeSelector` to add to the Deployment | `{}` |
+| `nodeplugin.tolerations` | List of Kubernetes `tolerations` to add to the Deployment | `{}` |
+| `nodeplugin.forcecephkernelclient` | Set to true to enable Ceph Kernel clients on kernel < 4.17 which support quotas | `true` |
+| `nodeplugin.podSecurityPolicy.enabled` | If true, create & use Pod Security [Policy resources](https://kubernetes.io/docs/concepts/policy/pod-security-policy/). | `false` |
+| `provisioner.name` | Specifies the name of provisioner | `provisioner` |
+| `provisioner.replicaCount` | Specifies the replicaCount | `3` |
+| `provisioner.timeout` | Timeout for waiting for creation or deletion of a volume | `60s` |
+| `provisioner.priorityClassName` | set user created priorityclassName for csi provisioner pods. default is system-cluster-critical which is less priority than system-node-critical | `system-cluster-critical` |
```
Comment (Yuggupta27) [inline charts/ceph-csi-cephfs/README.md]:
```suggestion
| `provisioner.priorityClassName` | Set user created priorityclassName for csi provisioner pods. default is system-cluster-critical which is less priority than system-node-critical | `system-cluster-critical` |
```

Comment (Yuggupta27) [reply charts/ceph-csi-cephfs/README.md]: Follow a similar format throughout the Description column. It looks much clear ;)

**Review by Yuggupta27 — CHANGES_REQUESTED** [review]
Follow similar suggestions for rbd too.

**Comment by Yuggupta27** [comment]
> @Yuggupta27 , #1786 adds extra options too. These need to be documented when this PR is merged. Maybe review this now, and rebase your PR once this is merged?

Sure @nixpanic. Since #1786 is just on the verge of getting merged along with https://github.com/ceph/ceph-csi/pull/2242 , I would prefer not to block that PR as it is up for quite some time. Since this is a doc change, once #1786 is merged, I can add a commit here on this PR to keep the doc changes at the same place. Wdyt?

File: charts/ceph-csi-cephfs/README.md [hunk 7]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `provisioner.affinity`                     | Specifies the affinity                                                                                                                           | `{}`                                               |
+| `provisioner.podSecurityPolicy.enabled`    | Specifies whether podSecurityPolicy is enabled                                                                                                   | `false`                                            |
+| `topology.enabled`                         | Specifies whether topology based provisioning support should be exposed by CSI                                                                   | `false`                                            |
+| `topology.domainLabels`                    | domainLabels define which node labels to use as domains for CSI nodeplugins to advertise their domains                                           | `{}`                                               |
```
Comment (Yuggupta27) [inline charts/ceph-csi-cephfs/README.md]:
```suggestion
| `topology.domainLabels`                    | DomainLabels define which node labels to use as domains for CSI nodeplugins to advertise their domains                                           | `{}`                                               |
```

File: charts/ceph-csi-rbd/README.md [hunk 8]
```diff
@@ -71,3 +71,78 @@ If you want to delete the namespace, use this command
...
+| `logLevel`                                 | Set logging level for csi containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                      | `5`                                                |
+| `nodeplugin.name`                          | Specifies the nodeplugins name                                                                                                                               | `nodeplugin`                                       |
+| `nodeplugin.updateStrategy`                | Specifies the update Strategy. If you are using ceph-fuse client set this value to OnDelete                                                      | `RollingUpdate`                                    |
+| `nodeplugin.priorityClassName`             | set user created priorityclassName for csi plugin pods. default is system-node-critical which is highest priority                                | `system-node-critical`                             |
```
Comment (Yuggupta27) [inline charts/ceph-csi-rbd/README.md]:
```suggestion
| `nodeplugin.priorityClassName`             | Set user created priorityclassName for csi plugin pods. default is system-node-critical which is highest priority                                | `system-node-critical`                             |
```

File: charts/ceph-csi-rbd/README.md [hunk 9]
```diff
@@ -71,3 +71,78 @@ If you want to delete the namespace, use this command
...
+| `nodeplugin.plugin.image.pullPolicy`       | Image pull policy                                                                                                                                | `IfNotPresent`                                     |
+| `nodeplugin.nodeSelector`                  | Kubernetes `nodeSelector` to add to the Deployment                                                                                               | `{}`                                               |
+| `nodeplugin.tolerations`                   | List of Kubernetes `tolerations` to add to the Deployment                                                                                        | `{}`                                               |
+| `nodeplugin.podSecurityPolicy.enabled`     | If true, create & use Pod Security [Policy resources](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).                          | `false`                                            |
```
Comment (Yuggupta27) [inline charts/ceph-csi-rbd/README.md]:
```suggestion
| `nodeplugin.podSecurityPolicy.enabled`     | If true, create & use [Pod Security Policy resources](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).                          | `false`                                            |
```

File: charts/ceph-csi-rbd/README.md [hunk 10]
```diff
@@ -71,3 +71,78 @@ If you want to delete the namespace, use this command
...
+| `provisioner.softMaxCloneDepth`            | Soft limit for maximum number of nested volume clones that are taken before a flatten occurs                                                     | `4`                                                |
+| `provisioner.maxSnapshotsOnImage`          | Maximum number of snapshots allowed on rbd image without flattening                                                                              | `450`                                              |
+| `provisioner.minSnapshotsOnImage`          | Minimum number of snapshots allowed on rbd image to trigger flattening                                                                           | `250`                                              |
+| `provisioner.skipForceFlatten`             | skip image flattening if kernel support mapping of rbd images which has the deep-flatten feature                                                 | `false`                                            |
```
Comment (Yuggupta27) [inline charts/ceph-csi-rbd/README.md]:
```suggestion
| `provisioner.skipForceFlatten`             | Skip image flattening if kernel support mapping of rbd images which has the deep-flatten feature                                                 | `false`                                            |
```

File: charts/ceph-csi-rbd/README.md [hunk 11]
```diff
@@ -71,3 +71,78 @@ If you want to delete the namespace, use this command
...
+| `provisioner.timeout`                      | Timeout for waiting for creation or deletion of a volume                                                                                         | `60s`                                              |
+| `provisioner.priorityClassName`            | Set user created priorityclassName for csi provisioner pods. default is system-cluster-critical which is less priority than system-node-critical | `system-cluster-critical`                          |
```
Comment (Yuggupta27) [inline charts/ceph-csi-rbd/README.md]:
```suggestion
| `provisioner.priorityClassName`            | Set user created priorityclassName for csi provisioner pods. Default is `system-cluster-critical` which is less priority than `system-node-critical` | `system-cluster-critical`                          |
```

**Review by Yuggupta27 — CHANGES_REQUESTED** [review]

File: charts/ceph-csi-cephfs/README.md [hunk 12]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `nodeplugin.updateStrategy`                | Specifies the update Strategy. If you are using ceph-fuse client set this value to OnDelete                                                      | `RollingUpdate`                                    |
+| `nodeplugin.priorityClassName`             | Set user created priorityclassName for csi plugin pods. default is system-node-critical which is highest priority                                | `system-node-critical`                             |
+| `nodeplugin.profiling.enabled`             | Specifies whether profiling should be enabled                                                                                                    | `false`                                            |
+| `nodeplugin.registrar.image.repository`    | Image Repository                                                                                                                                            | `k8s.gcr.io/sig-storage/csi-node-driver-registrar` |
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/README.md]: mention which image Repository? ex node-registrar image repo URL

File: charts/ceph-csi-cephfs/README.md [hunk 13]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `nodeplugin.plugin.image.tag`              | Image tag                                                                                                                                        | `canary`                                           |
+| `nodeplugin.plugin.image.pullPolicy`       | Image pull policy                                                                                                                                | `IfNotPresent`                                     |
+| `nodeplugin.nodeSelector`                  | Kubernetes `nodeSelector` to add to the Deployment                                                                                               | `{}`                                               |
+| `nodeplugin.tolerations`                   | List of Kubernetes `tolerations` to add to the Deployment                                                                                        | `{}`                                               |
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/README.md]: its daemonset not deployment?

File: charts/ceph-csi-cephfs/README.md [hunk 14]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `nodeplugin.podSecurityPolicy.enabled`     | If true, create & use [Pod Security Policy resources](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).                          | `false`                                            |
+| `provisioner.name`                         | Specifies the name of provisioner                                                                                                                | `provisioner`                                      |
+| `provisioner.replicaCount`                 | Specifies the replicaCount                                                                                                                       | `3`                                                |
+| `provisioner.timeout`                      | Timeout for waiting for creation or deletion of a volume                                                                                         | `60s`                                              |
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/README.md]: mention as GRPC timeout?

File: charts/ceph-csi-cephfs/README.md [hunk 15]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `provisioner.attacher.image.pullPolicy`    | Specifies pull policy                                                                                                                            | `IfNotPresent`                                     |
+| `provisioner.attacher.name`                | Specifies the name of csi-attacher sidecar                                                                                                                               | `attacher`                                         |
+| `provisioner.attacher.enabled`             | Specifies whether attacher sidecar is enabled                                                                                                    | `true`                                             |
+| `provisioner.resizer.image.repository`     | Specifies the csi-resizer sidecar                                                                                                                | `k8s.gcr.io/sig-storage/csi-resizer`               |
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/README.md]: keep it similar to another comment about repository

File: charts/ceph-csi-cephfs/README.md [hunk 16]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `provisioner.snapshotter.image.pullPolicy` | Specifies pull policy                                                                                                                            | `IfNotPresent`                                     |
+| `provisioner.nodeSelector`                 | Specifies the nodeSelector                                                                                                                       | `{}`                                               |
+| `provisioner.tolerations`                  | Specifies the tolerations                                                                                                                        | `{}`                                               |
+| `provisioner.affinity`                     | Specifies the affinity                                                                                                                           | `{}`                                               |
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/README.md]: mentiond about provisioner like ` Specify node selector  for provisioner deployment`

File: charts/ceph-csi-cephfs/README.md [hunk 17]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `pluginSocketFile`                         | The filename of the plugin socket                                                                                                                | `csi.sock`                                         |
+| `kubeletDir`                               | Kubelet working directory                                                                                                                        | `/var/lib/kubelet`                                 |
+| `driverName`                               | Name of the csi-driver                                                                                                                           | `cephfs.csi.ceph.com`                              |
+| `configMapName`                            | Name of the configmap used for state                                                                                                             | `ceph-csi-config`                                  |
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/README.md]: Name of the configmap which contains cluster configuration

File: charts/ceph-csi-cephfs/README.md [hunk 18]
```diff
@@ -71,3 +71,70 @@ If you want to delete the namespace, use this command
...
+| `externallyManagedConfigmap`               | Specifies the use of an externally provided configmap                                                                                            | `false`                                            |
+
+### Command Line
+
+You can pass the settings with helm command line parameters.
+Specify each parameter using the --set key=value argument to helm install.
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/README.md:140]: provide an example here?

**Review by Madhu-1 — COMMENTED** [review]
Reviewed cephfs same comments applicable for RBD also.

**Review by nixpanic — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**